pkg/alertmanager/statefulset.go

```diff
@@ -254,6 +273,13 @@ func makeStatefulSetSpec(logger *slog.Logger, a *monitoringv1.Alertmanager, conf
 		amArgs = append(amArgs, fmt.Sprintf("--enable-feature=%v", strings.Join(a.Spec.EnableFeatures[:], ",")))
 	}
 
+	if version.GTE(semver.MustParse("0.28.0")) {
```

Comment:
Is it necessary to determine whether the corresponding FeatureGate is enabled?

Reply:
Hi @dongjiang1989 changed the logic as Simon suggested to change to use monitoringv1.Argument instead so I'm resolving this. If you could review the changes then